proxy-server/background/proxy-handler.js

```diff
@@ -0,0 +1,69 @@
+// Log any errors from the proxy script
+browser.proxy.onProxyError.addListener(error => {
+  console.error(`Proxy error: ${error.message}`);
+});
+
+// Initialize the proxy
+function handleInit(message) {
+  // update the proxy whenever stored settings change
+  browser.storage.onChanged.addListener((newSettings) => {
+    if (newSettings.proxyServerInfo) {
+      console.log("Sending updated proxyServerInfo to proxy script");
+      browser.runtime.sendMessage(newSettings.proxyServerInfo.newValue, {toProxyScript: true});
```

Comment:
Since storage is updated whenever any of the options pages textareas change, the proxy script will temporarily get inconsistent and potentially incorrect information if the user updates, say, host and port together. I don't know how much this matters really, but it's technically a bug.

One way to resolve it would be to have an explicit "save" button in the options page, that updates storage with all the new settings together.